ignite/contrib/metrics/regression/wave_hedges_distance.py

```diff
@@ -0,0 +1,26 @@
+from __future__ import division
+import torch
+from ignite.metrics.metric import Metric
+
+
+class WaveHedgesDistance(Metric):
+    r"""
+    Calculates the Wave Hedges Distance.
+    It has been proposed in `Botchkarev 2018`__.
+    :math:`\text{WHD} = \sum _j^n\frac{|A_j - P_j|}{max(A_j, P_j)}`
+    Where, :math:`A_j` is the ground truth and :math:`P_j` is the predicted value.
+    - `update` must receive output of the form `(y_pred, y)`.
+    - `y` and `y_pred` must be of same shape.
+    __ https://arxiv.org/abs/1809.03006
+    """
+
+    def reset(self):
+        self._sum_of_errors = 0.0
+
+    def update(self, output):
+        y_pred, y = output
+        errors = torch.abs(y.view_as(y_pred) - y_pred) / torch.max(y_pred, y.view_as(y_pred))
```

Comment:
I have a doubt on this one, I'm not sure that $max(A_j, P_j)$ = `torch.max(y_pred, y.view_as(y_pred))`. Need to check this from my side

Reply:
@vfdev-5 see code below

```python
y_pred = torch.rand(10)
y = torch.rand(10)
maxes = torch.max(y_pred, y)

print (y_pred)
# tensor([0.1823, 0.0923, 0.1578, 0.9375, 0.4769, 0.0928, 0.7895, 0.2848, 0.4338, 0.2930])
print (y)
# tensor([0.3095, 0.8486, 0.2325, 0.3083, 0.6410, 0.8848, 0.5019, 0.0213, 0.1661, 0.2154])
print (maxes)
# tensor([0.3095, 0.8486, 0.2325, 0.9375, 0.6410, 0.8848, 0.7895, 0.2848, 0.4338, 0.2930])
```

Reply:
You are right, `torch.max` is element-wise max if two tensors are compared.